parts/k8s/hostsconfigagent.ps1

```diff
@@ -0,0 +1,59 @@
+$Global:ClusterConfiguration = ConvertFrom-Json ((Get-Content "c:\k\kubeclusterconfig.json" -ErrorAction Stop) | out-string)
+$clusterFQDN = $Global:ClusterConfiguration.Kubernetes.ControlPlane.IpAddress
+$hostsFile="C:\Windows\System32\drivers\etc\hosts"
+$retryDelaySeconds = 15
+
+# TODO: test only, remove it after test
+$clusterFQDN = "$clusterFQDN.local"
+
+# Dot-source scripts with functions that are called in this script
+. c:\AzureData\k8s\kuberneteswindowsfunctions.ps1
+
+function Get-APIServer-IPAddress
+{
+    $uri = "http://169.254.169.254/metadata/instance/compute/tags?api-version=2019-03-11&format=text"
+    $response = Retry-Command -Command "Invoke-RestMethod" -Args @{Uri=$uri; Method="Get"; ContentType="application/json"; Headers=@{"Metadata"="true"}} -Retries 3 -RetryDelaySeconds 5
+
+    if(!$response) {
+        return ""
+    }
+
+    foreach ($tag in $response.Split(";"))
+    {
+        $values = $tag.Split(":")
+        if ($values.Length -ne 2)
+        {
+            return ""
+        }
+
+        if ($values[0] -eq "aksAPIServerIPAddress")
+        {
+            return $values[1]
+        }
+    }
+
+    return ""
+}
+
+while ($true)
```

Comment:
this looks like an indefinite loop.  Is that the intent?

Reply:
I see below this file is installed as a service to continually poll

Reply:
exactly